var/spack/repos/builtin/packages/py-dace/package.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyDace(PythonPackage):
+    """DaCe is a fast parallel programming framework that takes code in
+    Python/NumPy and other programming languages, and maps it to
+    high-performance CPU, GPU, and FPGA programs, which can be
+    optimized programmatically or interactively."""
+
+    homepage = "https://github.com/spcl/dace"
+    pypi = "dace/dace-0.14.2.tar.gz"
+    git = "https://github.com/spcl/dace.git"
+
+    maintainers("tbennun")
+
+    version("master", branch="master", submodules=True)
+    version("0.14.2", sha256="13e5c5af31ca7313839dee257305d2020ac79c3cfa77af8ea3feb5260b0582a4")
+
+    variant(
+        "counters",
+        description="Optional requirements that enable performance counter collection.",
+        default=False,
+    )
+
+    depends_on("python@3.7:", when="@0.14.2:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    # From setup.py
+    depends_on("cmake@3.15.0:", when="@0.14.2:", type=("build", "run"))
```

Comment:
Missing `py-scikit-build` dependency. Technically, dependency should be `py-cmake`, not `cmake`.

Reply:
Not necessarily. The Python cmake dependency ensures the existence of CMake (I don't think py-cmake has any Python API). If an existing CMake is found, the `setup.py` script will not look for py-cmake: https://github.com/spcl/dace/blob/v0.14.2/setup.py#L35